kpops/components/base_components/kafka_app.py

```diff
@@ -110,13 +110,28 @@ async def clean(self, dry_run: bool) -> None:
         log.info(f"Uninstall old cleanup job for {self.helm_release_name}")
         await self.destroy(dry_run)
 
+        image_tag = await self.fetch_image_tag()
+        self.app.image_tag = image_tag
+
         log.info(f"Init cleanup job for {self.helm_release_name}")
         await self.deploy(dry_run)
 
         if not self.config.retain_clean_jobs:
             log.info(f"Uninstall cleanup job for {self.helm_release_name}")
             await self.destroy(dry_run)
 
+    async def fetch_image_tag(self) -> str:
+        """Fetch the image tag of the streams-bootstrap app using the 'helm get values' command.
+
+        If the release doesn't exist, it will fall back to the specified imageTag in the default.yaml/pipeline.yaml
+
+        :return: Image tag of the streams-bootstrap app
+        """
+        cluster_values = await self.helm.get_values(
+            self.namespace, self.helm_release_name
+        )
+        return cluster_values["imageTag"] if cluster_values else self.app.image_tag
```

Comment:
how about deserializing it to `StreamsBootstrapValues`?

```suggestion
if cluster_values:
    return StreamsBootstrapValues.model_validate(cluster_values).image_tag
```

I think I would also prefer returning an optional here and let the caller handle what happens in the `None` case

Reply:
Addressed

Reply:
@raminqaf did you forget to push? 😄

Reply:
I have changed the logic a bit. So this comment becomes obsolete :)